### PR TITLE
Remove unused `LocalChannel` class in the router

### PR DESCRIPTION
**Pull request by pm47**
Builds on #2264.

### CODE REVIEW

File: eclair-core/src/test/scala/fr/acinq/eclair/integration/PaymentIntegrationSpec.scala [hunk 1]
```diff
@@ -127,6 +127,8 @@ class PaymentIntegrationSpec extends IntegrationSpec {
           sender.expectMsgType[Iterable[ChannelAnnouncement]].size == channels
         }, max = 60 seconds, interval = 1 second)
         awaitCond({
+
+
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/integration/PaymentIntegrationSpec.scala:131]: nit: why the extra lines?

File: eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala [hunk 2]
```diff
@@ -368,24 +362,14 @@ object Router {
...
-      case c: PublicChannel => if (c.ann.nodeId1 == localNodeId) c.ann.nodeId2 else c.ann.nodeId1
-    }
-    /** Our remote peer's channel_update: this is what must be used in invoice routing hints. */
-    val remoteUpdate: Option[ChannelUpdate] = channel match {
-      case c: PrivateChannel => if (remoteNodeId == c.nodeId1) c.update_1_opt else c.update_2_opt
-      case c: PublicChannel => if (remoteNodeId == c.ann.nodeId1) c.update_1_opt else c.update_2_opt
-    }
     /** Create an invoice routing hint from that channel. Note that if the channel is private, the invoice will leak its existence. */
-    def toExtraHop: Option[ExtraHop] = remoteUpdate.map(u => ExtraHop(remoteNodeId, u.shortChannelId, u.feeBaseMsat, u.feeProportionalMillionths, u.cltvExpiryDelta))
+    def toExtraHop: Option[ExtraHop] = {
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala:366]: nit: I find the name a bit confusing, it doesn't really convey that we'll take the remote direction...maybe `toIncomingExtraHop` would be better?

**Review by t-bast — COMMENTED** [review]
LGTM

File: eclair-core/src/test/scala/fr/acinq/eclair/integration/PaymentIntegrationSpec.scala (hunk 1 above)
Comment (pm47) [reply eclair-core/src/test/scala/fr/acinq/eclair/integration/PaymentIntegrationSpec.scala:131]: I'm cherry picking individual changes from my prototype branch, some artefacts go through 🤷‍♂️

Comment (pm47) [reply eclair-core/src/test/scala/fr/acinq/eclair/integration/PaymentIntegrationSpec.scala]: Fixed in 35bfff8f3.

File: eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala (hunk 2 above)
Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala]: Done in e3f64d2eb.

**Review by t-bast — APPROVED** [review]

**Pull request merged**